src/tools/analyzer/sss_analyze.py

```diff
@@ -1,27 +1,166 @@
 #!/usr/bin/env python
 
-import click
+import argparse
 
 import source_files
 
 from modules import request
 
 
-class Analyzer(object):
-    def __init__(self, source="files", logdir="/var/log/sssd/"):
-        self.source = source
-        self.logdir = logdir
+# Based on patch from https://bugs.python.org/issue9341
+class SubparsersAction(argparse._SubParsersAction):
+    """
+    Provide a subparser action that can create subparsers with ability of
+    grouping arguments.
 
+    It is based on the patch from:
 
-@click.group(help="Analyzer tool to assist with SSSD Log parsing")
-@click.option('--source', default='files')
-@click.option('--logdir', default='/var/log/sssd/', help="SSSD Log directory "
-              "to parse log files from")
-@click.pass_context
-def cli(ctx, source, logdir):
-    ctx.obj = Analyzer(source, logdir)
+        - https://bugs.python.org/issue9341
+    """
+
+    class _PseudoGroup(argparse.Action):
+
```

Comment:
nitpicking - Looking at the overall style this empty line should be avoided.

Reply:
Thanks, fixed - and rebased PR.